File: EESoft/DataCenter/DataCenter - Server/Forms/EditUsersFrm.cs
using Microsoft.VisualBasic;
using System;

namespace DataCenter_Server
{
	public partial class EditUsersFrm
	{
		public EditUsersFrm()
		{
			InitializeComponent();
		}
		private string ErrorProvider;
		int Int1;

        public void UsersFrm_Load(Object sender, EventArgs e)
        {
            ErrorProvider = MainModule.FunLib.GetAllUsers(MainModule.AdminID, true);
            ErrorProvider = null;
            if (ErrorProvider == null && MainModule.FunLib.AllUsers != null)
            {
                LBUsers.Items.Clear();
                if (MainModule.FunLib.AllUsers.Length <= 0)
                    return;
                for (Int1 = 0; Int1 <= MainModule.FunLib.AllUsers.Length - 1; Int1++)
                    LBUsers.Items.Add(MainModule.FunLib.AllUsers[Int1].UserName);
            }
            else
            {
                MainModule.MsgBox_(ErrorProvider, (short)0, (short)20, false, this);
                Close();
            }
        }
        public void TxtNewUser_TextChanged(Object sender, EventArgs e)
        {
            if (TxtNewUser.Text.Trim().Length > 0)
                BtnAdd.Enabled = true;
            else
                BtnAdd.Enabled = false;
        }
        public void LBUsers_SelectedIndexChanged(Object sender, EventArgs e)
        {
            if (LBUsers.SelectedIndex != -1)
            {
                BtnRemove.Enabled = true;
                BtnReset.Enabled = true;
                TxtUpdate.ReadOnly = false;
                TxtUpdate.Text = (string)LBUsers.SelectedItem;
            }
            else
            {
                BtnRemove.Enabled = false;
                BtnReset.Enabled = false;
                TxtUpdate.Clear();
                TxtUpdate.ReadOnly = true;
            }
        }
        public void TxtUpdate_TextChanged(Object sender, EventArgs e)
        {
            if (TxtUpdate.Text.Trim().Length > 0)
                BtnUpdate.Enabled = true;
            else
                BtnUpdate.Enabled = false;
        }
        public void BtnAdd_Click(Object sender, EventArgs e)
        {
            ErrorProvider = "";
            ErrorProvider = MainModule.FunLib.AddUser(TxtNewUser.Text.Trim());
            if (ErrorProvider != "UserName In Used")
            {
                if (MainModule.FunLib.AllUsers == null)
                {
                    Array.Resize(ref MainModule.FunLib.AllUsers, 1);
                }
                else
                {
                    Array.Resize(ref MainModule.FunLib.AllUsers, MainModule.FunLib.AllUsers.Length + 1);
                }
                MainModule.FunLib.AllUsers[MainModule.FunLib.AllUsers.Length - 1].UsrID = ErrorProvider;
                MainModule.FunLib.AllUsers[MainModule.FunLib.AllUsers.Length - 1].UserName = TxtNewUser.Text.Trim();
                LBUsers.Items.Add(TxtNewUser.Text.Trim());
                MainModule.MsgBox_("تم الاضافه", 88, 23, true, this);
            }
            else
            {
                MainModule.MsgBox_(ErrorProvider, 0, 20, false, this);
            }
        }
        public void BtnUpdate_Click(Object sender, EventArgs e)
        {
            if (Interaction.MsgBox("هل انت متأكد؟", MsgBoxStyle.YesNo, "تحزير") == MsgBoxResult.Yes)
            {
                if (LBUsers.SelectedIndex != -1 && TxtUpdate.Text.Trim().Length > 0)
                {
                    ErrorProvider = null;
                    ErrorProvider = MainModule.FunLib.UpdateUserameOrPassword(MainModule.FunLib.AllUsers[LBUsers.SelectedIndex].UsrID, TxtUpdate.Text.Trim(), true);
                    if (ErrorProvider == null)
                    {
                        MainModule.FunLib.AllUsers[LBUsers.SelectedIndex].UserName = TxtUpdate.Text.Trim();
                        LBUsers.Items[LBUsers.SelectedIndex] = TxtUpdate.Text.Trim();
                        MainModule.MsgBox_("تم التعديل", 77, 23, true, this);
                    }
                    else
                        MainModule.MsgBox_("لم يتم التعديل", 98, 23, false, this);
                }
            }
        }
        public void BtnReset_Click(Object sender, EventArgs e)
        {
            if (Interaction.MsgBox("هل انت متأكد؟", MsgBoxStyle.YesNo, "تحزير") == MsgBoxResult.Yes)
            {
                if (LBUsers.SelectedIndex != -1 && TxtUpdate.Text.Trim().Length > 0)
                {
                    ErrorProvider = null;
                    ErrorProvider = MainModule.FunLib.UpdateUserameOrPassword(MainModule.FunLib.AllUsers[LBUsers.SelectedIndex].UsrID, MainModule.DefaultUserPassword, false);
                    if (ErrorProvider == null)
                    {
                        MainModule.FunLib.AllUsers[LBUsers.SelectedIndex].UserName = TxtUpdate.Text.Trim();
                        MainModule.MsgBox_("تم التعديل", 77, 23, true, this);
                    }
                    else
                        MainModule.MsgBox_("لم يتم التعديل", 98, 23, false, this);
                }
            }
        }
        public void BtnRemove_Click(Object sender, EventArgs e)
        {
            if (Interaction.MsgBox("هل انت متأكد؟", MsgBoxStyle.YesNo, "تحزير") == MsgBoxResult.Yes)
            {
                if (LBUsers.SelectedIndex != -1)
                {
                    ErrorProvider = null;
                    ErrorProvider = (string)MainModule.FunLib.UserActivation(MainModule.FunLib.AllUsers[LBUsers.SelectedIndex].UsrID, false);
                    if (ErrorProvider == null)
                    {
                        UsersFrm_Load(this, e);
                        BtnRemove.Enabled = false;
                        TxtUpdate.ReadOnly = true;
                        BtnUpdate.Enabled = false;
                        MainModule.MsgBox_("تم الحذف", (short)125, (short)23, true, this);
                    }
                    else
                        MainModule.MsgBox_("لم يتم الحذف", (short)111, (short)23, false, this);

                }
            }
        }
	}
}

File: EESoft/DataCenterVB/Backup/DataCenter - Server/Forms/EditUsersFrm.vb
Public Class EditUsersFrm
    Private ErrorProvider As String = Nothing
    Dim Int1 As Integer

    Private Sub UsersFrm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ErrorProvider = FunLib.GetAllUsers(AdminID, True)
        ErrorProvider = Nothing
        If ErrorProvider Is Nothing And Not FunLib.AllUsers Is Nothing Then
            LBUsers.Items.Clear()
            If FunLib.AllUsers.Length <= 0 Then
                Exit Sub
            End If
            For Int1 = 0 To FunLib.AllUsers.Length - 1
                LBUsers.Items.Add(FunLib.AllUsers(Int1).UserName)
            Next
        Else
            'Prepare MSG
            MsgBox_(ErrorProvider, 0, 20, False, Me)
            Me.Close()
        End If
    End Sub
    Private Sub TxtNewUser_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TxtNewUser.TextChanged
        If TxtNewUser.Text.Trim.Length > 0 Then
            BtnAdd.Enabled = True
        Else
            BtnAdd.Enabled = False
        End If
    End Sub
    Private Sub LBUsers_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles LBUsers.SelectedIndexChanged
        If LBUsers.SelectedIndex <> -1 Then
            BtnRemove.Enabled = True
            BtnReset.Enabled = True
            TxtUpdate.ReadOnly = False
            TxtUpdate.Text = LBUsers.SelectedItem
        Else
            BtnRemove.Enabled = False
            BtnReset.Enabled = False
            TxtUpdate.Clear()
            TxtUpdate.ReadOnly = True
        End If
    End Sub
    Private Sub TxtUpdate_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TxtUpdate.TextChanged
        If TxtUpdate.Text.Trim.Length > 0 Then
            BtnUpdate.Enabled = True
        Else
            BtnUpdate.Enabled = False
        End If
    End Sub
    Private Sub BtnAdd_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnAdd.Click
        ErrorProvider = ""
        ErrorProvider = FunLib.AddUser(TxtNewUser.Text.Trim)
        If ErrorProvider <> "UserName In Used" Then
            If FunLib.AllUsers Is Nothing Then
                ReDim FunLib.AllUsers(0)
            Else
                ReDim Preserve FunLib.AllUsers(FunLib.AllUsers.Length)
            End If
            FunLib.AllUsers(FunLib.AllUsers.Length - 1).UsrID = ErrorProvider
            FunLib.AllUsers(FunLib.AllUsers.Length - 1).UserName = TxtNewUser.Text.Trim
            LBUsers.Items.Add(TxtNewUser.Text.Trim)
            MsgBox_("New user added successfully", 88, 23, True, Me)
        Else
            MsgBox_(ErrorProvider, 0, 20, False, Me)
        End If
    End Sub
    Private Sub BtnUpdate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnUpdate.Click
        If MsgBox("Are you sure ?", MsgBoxStyle.YesNo, "WARNING") = MsgBoxResult.Yes Then
            If LBUsers.SelectedIndex <> -1 And TxtUpdate.Text.Trim.Length > 0 Then
                ErrorProvider = Nothing
                ErrorProvider = FunLib.UpdateUserameOrPassword(FunLib.AllUsers(LBUsers.SelectedIndex).UsrID, TxtUpdate.Text.Trim, True)
                If ErrorProvider Is Nothing Then
                    FunLib.AllUsers(LBUsers.SelectedIndex).UserName = TxtUpdate.Text.Trim
                    LBUsers.Items.Item(LBUsers.SelectedIndex) = TxtUpdate.Text.Trim
                    MsgBox_("Username changed successfully", 77, 23, True, Me)
                Else
                    MsgBox_("Username didn't changed", 98, 23, False, Me)
                End If
            End If
        End If
    End Sub
    Private Sub BtnReset_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnReset.Click
        If MsgBox("Are you sure ?", MsgBoxStyle.YesNo, "WARNING") = MsgBoxResult.Yes Then
            If LBUsers.SelectedIndex <> -1 And TxtUpdate.Text.Trim.Length > 0 Then
                ErrorProvider = Nothing
                ErrorProvider = FunLib.UpdateUserameOrPassword(FunLib.AllUsers(LBUsers.SelectedIndex).UsrID, DefaultUserPassword, False)
                If ErrorProvider Is Nothing Then
                    FunLib.AllUsers(LBUsers.SelectedIndex).UserName = TxtUpdate.Text.Trim
                    MsgBox_("Password changed successfully", 77, 23, True, Me)
                Else
                    MsgBox_("Password didn't changed", 98, 23, False, Me)
                End If
            End If
        End If
    End Sub
    Private Sub BtnRemove_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnRemove.Click
        If MsgBox("Are you sure ?", MsgBoxStyle.YesNo, "WARNING") = MsgBoxResult.Yes Then
            If LBUsers.SelectedIndex <> -1 Then
                ErrorProvider = Nothing
                ErrorProvider = FunLib.UserActivation(FunLib.AllUsers(LBUsers.SelectedIndex).UsrID, False)
                If ErrorProvider = Nothing Then
                    UsersFrm_Load(sender, e)
                    BtnRemove.Enabled = False
                    TxtUpdate.ReadOnly = True
                    BtnUpdate.Enabled = False
                    MsgBox_("User Disactivated", 125, 23, True, Me)
                Else
                    MsgBox_("Can't Disactivate user", 111, 23, False, Me)
                End If
            End If
        End If
    End Sub
End Class